core/scripts/chaincli/command/keeper/upkeep.go

```diff
@@ -29,3 +35,57 @@ var upkeepCmd = &cobra.Command{
 		hdlr.UpkeepCounterEvents(cmd.Context(), args[0], fromBlock, toBlock)
 	},
 }
+
+// upkeepHistoryCmd represents the command to run the upkeep history command
+var upkeepHistoryCmd = &cobra.Command{
+	Use:   "upkeep-history",
+	Short: "Print checkUpkeep history",
+	Long:  `Print checkUpkeep status and keeper responsibility for a given upkeep in a set block range`,
+	Run: func(cmd *cobra.Command, args []string) {
+		fromBlock, toBlock := getRange(cmd)
+		upkeepId, err := cmd.Flags().GetInt64("upkeep-id")
+		if err != nil {
+			log.Fatal("failed to get upkeep-id flag: ", err)
+		}
+
+		cfg := config.New()
+		hdlr := handler.NewKeeper(cfg)
+
+		hdlr.UpkeepHistory(cmd.Context(), upkeepId, fromBlock, toBlock)
+	},
+}
+
+func getRange(cmd *cobra.Command) (uint64, uint64) {
+	rangeRaw, err := cmd.Flags().GetString("range")
```

Comment:
Why not take two separate int arguments to remove complexity of regex and parsing

Reply:
Makes sense, done